enumerable.rb

```diff
@@ -0,0 +1,119 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+    i = 0
+    while i < self.length
+      yield(self[i])
+      i += 1     
+    end
+    self
+  end
+  
+  def my_each_with_index
+    return enum_for unless block_given?
+    i =0
+    while i < self.length
+      yield(self[i], i)
+      i += 1   
+    end
+    self
+  end
+
+  def my_select
```

Comment:
block = proc { |num| num < (0 + 9 ) / 2 } }
Testing Range(5, 50).select(&block) is not equal to Range(5,50).my_select(&block)